src/H5TSint.c

```diff
@@ -556,18 +568,30 @@ H5TS__tinfo_term(void)
         HGOTO_DONE(FAIL);
     while (H5TS_tinfo_next_free_s) {
         H5TS_tinfo_node_t *next = H5TS_tinfo_next_free_s->next;
+
+        /* Free any entries within the error stack */
+        H5E__clear_stack(&H5TS_tinfo_next_free_s->info.err_stack);
```

Comment:
Two problems: there are no routines that can push errors on the stack, and I am removing all the calls to H5E_clear_stack from the library (except for within the error code and the single call in the FUNC_LEAVE_* macros.

Reply:
Are you saying it's a problem because we can't push any errors in the event that this fails, or that this routine is unnecessary because no errors can be pushed here? If it's the latter, this is meant to clean up errors that were pushed before thread termination (see `ttsafe_error_stacks.c`). If the former, I would assume that returning `FAIL` if this fails should be sufficient, same as the rest of the termination routine. (I'll add a check for that.)

To your second point - I agree that's a problem! We may need to discuss that at greater length.

Reply:
I meant the latter.   But, if there's any errors on the default error stack, it's the H5E package's responsibility to clear them, not the H5TS package.

Reply:
These errors are in the thread-local error stacks from other threads. H5E could clear them itself, but that would require exposing more H5TS functionality to H5E.